shm/Cargo.hpp

```diff
@@ -0,0 +1,29 @@
+#ifndef CARGO_HPP
+#define CARGO_HPP
+#include <string>
+#include <memory>
+#include "TimeObserver.hpp"
+
+// Interface class used to represent cargo carried by each ship
+class Cargo : public TimeObserver
+{
+protected:
```

Comment:
better private. You have setter and getter so you can use them in derivered class